black_it/loss_functions/msm.py

```diff
@@ -103,12 +107,14 @@ def _validate_covariance_and_calculator(
                 is not 18. Other possible errors won't be caught by this
                 function, and can only be detected at runtime.
         """
-        if covariance_mat is None:
-            # if we were given no covariance matrix, then we'll use a default
-            # one, and we can't do any further validation (without executing the
-            # moment_calculator)
+        if covariance_mat in ("identity", "inverse_variance"):
+            # if we were given the covariance matrix as a string, we don't need any further validation
             return
-
+        if isinstance(covariance_mat, str):
+            raise ValueError(
+                "please specify a valid covariance matrix. Supported options are: 'identity', 'inverse_variance', "
+                "or directly a numpy array."
+            )
```

Comment:
at this point it is probably missing a check on the type of `covariance_mat`? Below we assume that if `covariance_mat` is not a `str`, then it is a `NDArray`, but this is not enforced anywhere. On one hand, this follows the Python philosophy ["we are all consenting adults"](https://python-guide-chinese.readthedocs.io/zh_CN/latest/writing/style.html#we-are-all-consenting-adults) and the user is responsible of what he's doing. Perhaps a one-more check is not a bad idea though.


```suggestion
            )
        if not isinstance(covariance_mat, np.ndarray):
            raise ValueError(...)
```

Reply:
thank you, I added this check